### PR TITLE
Adding QTUM read functions

### PR DESCRIPTION
**Pull request by Rachit-gaur**

### CODE REVIEW

File: src/blockchain/qtum.ts [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+export const getQtumTransactions = async (address: string,pageNum:number): Promise<QtumIRawTransactions>=>get(`/v3/qtum/transactions/address/${address}/${pageNum}`)
+
+export const getInfo = async (address: string): Promise<QtumIGetInfo> => get(`/v3/qtum/${address}/balance`);
+
+export const estimateFee = async (nblocks: number): Promise<any> => get(`/v3/qtum/transactions/gas/${nblocks}`);
+export const estimateFeePerByte = async (nblocks: number): Promise<any> => get(`/v3/qtum/transactions/gasbytes/${nblocks}`);
+
+export const broadcast = async (rawtx: string): Promise<QtumISendRawTxResult> => post('v3/qtum/broadcast', { rawtx });
+
+export const smartContract = async (address: string,encodedData: string): Promise<QtumIContractCall> => post('v3/qtum/smartcontract',{address,encodedData})
```
Comment (samuelsramko) [inline src/blockchain/qtum.ts:16]: Remove smart contract call endpoint pls. I dont want this one here yet.

File: src/wallet/address.ts [hunk 2]
```diff
@@ -249,6 +256,13 @@ const generateTronPrivateKey = async (mnemonic: string, i: number) => {
         .derive(i)
         .privateKey?.toString('hex') ?? '';
 };
+const generateQtumPrivateKey = async (testnet: boolean, mnem: string,i:number) => {
+    const network=testnet?QTUM_NETWORK_TESTNET:QTUM_NETWORK_MAINNET;
+    const hdwallet = HDKey.fromMasterSeed(await mnemonicToSeed(mnem), network.bip32);
+    console.log(hdwallet.derive(testnet ? TESTNET_DERIVATION_PATH : BTC_DERIVATION_PATH).toJSON())
+    return hdwallet.derive(testnet ? TESTNET_DERIVATION_PATH : BTC_DERIVATION_PATH).toJSON().xpub
```
Comment (samuelsramko) [inline src/wallet/address.ts:263]: You have btc path here

File: src/constants.ts [hunk 3]
```diff
@@ -17,6 +17,7 @@ export const VET_DERIVATION_PATH = 'm/44\'/818\'/0\'/0';
 export const BTC_DERIVATION_PATH = 'm/44\'/0\'/0\'/0';
 export const TRON_DERIVATION_PATH = 'm/44\'/195\'/0\'/0';
 export const LTC_DERIVATION_PATH = 'm/44\'/2\'/0\'/0';
```
Comment (samuelsramko) [inline src/constants.ts:19]: This path is Not complete as otjers

File: src/wallet/address.ts [hunk 4]
```diff
@@ -480,7 +494,11 @@ const convertBtcPrivateKey = (testnet: boolean, privkey: string) => {
     const keyPair = ECPair.fromWIF(privkey, network);
     return payments.p2pkh({pubkey: keyPair.publicKey, network}).address as string;
 };
-
+const convertQTUMPrivateKey = (testnet: boolean, privkey: string) => {
+    const network = testnet ? QTUM_NETWORK_TESTNET : QTUM_NETWORK_MAINNET;
+    const keyPair = ECPair.fromWIF(privkey, network);
+    return payments.p2pkh({pubkey: keyPair.publicKey, network}).address as string;
```
Comment (samuelsramko) [inline src/wallet/address.ts:500]: Is it possible to use segwit address here, not p2pkh? Does qtum supports that?

File: src/wallet/wallet.ts [hunk 5]
```diff
@@ -186,6 +187,13 @@ export const generateBtcWallet = async (testnet: boolean, mnem: string): Promise
         xpub: hdwallet.derive(testnet ? TESTNET_DERIVATION_PATH : BTC_DERIVATION_PATH).toJSON().xpub
     };
 };
+export const generateQtumWallet = async (testnet: boolean, mnem: string): Promise<Wallet> => {
+    const hdwallet = hdkey.fromMasterSeed(await mnemonicToSeed(mnem), testnet ? networks.testnet.bip32 : networks.bitcoin.bip32);
+    return {
+        mnemonic: mnem,
+        xpub: hdwallet.derive(QTUM_DERIVATION_PATH).toJSON().xpub
```
Comment (samuelsramko) [inline src/wallet/wallet.ts:194]: Missing testnet derivstion path

**Review by samuelsramko — CHANGES_REQUESTED** [review]

File: src/wallet/address.ts (hunk 4 above)
Comment (Rachit-gaur) [reply src/wallet/address.ts:500]: as discussed using p2pkh

File: src/wallet/address.ts [hunk 6]
```diff
@@ -249,6 +256,13 @@ const generateTronPrivateKey = async (mnemonic: string, i: number) => {
         .derive(i)
         .privateKey?.toString('hex') ?? '';
 };
+const generateQtumPrivateKey = async (testnet: boolean, mnem: string,i:number) => {
+    const network=testnet?QTUM_NETWORK_TESTNET:QTUM_NETWORK_MAINNET;
+    const hdwallet = HDKey.fromMasterSeed(await mnemonicToSeed(mnem), network.bip32);
+    console.log(hdwallet.derive(testnet ? TESTNET_DERIVATION_PATH : BTC_DERIVATION_PATH).toJSON())
```
Comment (samuelsramko) [inline src/wallet/address.ts:262]: BAsed on this. -https://github.com/satoshilabs/slips/blob/master/slip-0044.md use derivation path m/44'/2301'/0'/0

File: src/blockchain/qtum.ts [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+export const getQtumTransaction = async (id: string): Promise<QtumIRawTransactionInfo>=>get(`/v3/qtum/transaction/${id}`)
+export const getQtumTransactions = async (address: string,pageNum:number): Promise<QtumIRawTransactions>=>get(`/v3/qtum/transactions/address/${address}/${pageNum}`)
+
+export const getInfo = async (address: string): Promise<QtumIGetInfo> => get(`/v3/qtum/${address}/balance`);
+
+export const estimateFee = async (nblocks: number): Promise<any> => get(`/v3/qtum/transactions/gas/${nblocks}`);
+export const estimateFeePerByte = async (nblocks: number): Promise<any> => get(`/v3/qtum/transactions/gasbytes/${nblocks}`);
+
+export const broadcast = async (rawtx: string): Promise<QtumISendRawTxResult> => post('v3/qtum/broadcast', { rawtx });
+
```
Comment (samuelsramko) [inline src/blockchain/qtum.ts:15]: remove the smartContract endpoint

**Review by samuelsramko — CHANGES_REQUESTED** [review]

File: src/blockchain/qtum.ts [hunk 8]
```diff
@@ -0,0 +1,13 @@
+import { get, post } from '../connector/tatum'
+import { QtumIRawTransactions, QtumIRawTransactionInfo, QtumIContractCall, QtumISendRawTxResult } from "src/model/response/qtum/qtumTx"
+import { QtumIGetInfo, } from 'src/model/response/qtum/qtumInfo';
+import { QtumIUTXO } from 'src/model/response/qtum/qtumUTXO';
+export const getQtumUTXOs = async (address: string): Promise<QtumIUTXO> => get(`/v3/qtum/utxo/${address}`);
+export const getQtumTransaction = async (id: string): Promise<QtumIRawTransactionInfo> => get(`/v3/qtum/transaction/${id}`)
+export const getQtumTransactions = async (address: string, pageNum: number): Promise<QtumIRawTransactions> => get(`/v3/qtum/transactions/address/${address}/${pageNum}`)
+
```
Comment (samuelsramko) [inline src/blockchain/qtum.ts:8]: Missing getBlock and getCurrentBlock operations

File: src/wallet/wallet.ts [hunk 9]
```diff
@@ -186,6 +187,13 @@ export const generateBtcWallet = async (testnet: boolean, mnem: string): Promise
         xpub: hdwallet.derive(testnet ? TESTNET_DERIVATION_PATH : BTC_DERIVATION_PATH).toJSON().xpub
     };
 };
+export const generateQtumWallet = async (testnet: boolean, mnem: string): Promise<Wallet> => {
+    const hdwallet = hdkey.fromMasterSeed(await mnemonicToSeed(mnem), testnet ? networks.testnet.bip32 : networks.bitcoin.bip32);
```
Comment (samuelsramko) [inline src/wallet/wallet.ts:191]: Should not be here Qtum networks?

**Review by samuelsramko — CHANGES_REQUESTED** [review]

File: src/wallet/wallet.ts (hunk 9 above)
Comment (Rachit-gaur) [reply src/wallet/wallet.ts:191]: this is just interface, 
export declare const bitcoin: Network;
export interface Network {
    messagePrefix: string;
    bech32: string;
    bip32: Bip32;
    pubKeyHash: number;
    scriptHash: number;
    wif: number;
}

Comment (samuelsramko) [reply src/wallet/wallet.ts:191]: it's not just an interface. the xpub format is based on that values, so there must be QTUM network passed here correctly.

**Review by samuelsramko — CHANGES_REQUESTED** [review]

File: src/model/response/qtum/QtumBlock.ts [hunk 10]
```diff
@@ -0,0 +1,38 @@
+/**
+ *
+ * @export
+ * @interface QtumBlock
+ */
+export interface QtumBlock {
+    result: {
```
Comment (samuelsramko) [inline src/model/response/qtum/QtumBlock.ts:7]: no result and error. Please move the data like hash to the highest hierarchy.

Comment (Rachit-gaur) [reply src/model/response/qtum/QtumBlock.ts]: ok

**Review by samuelsramko — APPROVED** [review]

**Pull request merged**